pkg/kv/kvserver/kvflowcontrol/rac2/log_tracker.go

```diff
@@ -109,7 +110,7 @@ type LogTracker struct {
 	//	- waiting[pri][i].Term <= last.Term
 	//	- waiting[pri][i].Index < waiting[pri][i+1].Index
 	//	- waiting[pri][i].Term <= waiting[pri][i+1].Term
-	waiting [raftpb.NumPriorities]CircularBuffer[LogMark]
```

Comment:
just a drive-by that you don't have to address but something for "next time": I would prefer to see this in separate commits. As is, the commit tangles extracting the `ring.Buffer` with adopting it in `RawNode`.